components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -236,6 +238,24 @@ private String fetchHostName() throws URLBuilderException {
         return hostName;
     }
 
+    private String fetchInternalHostName() throws URLBuilderException {
+
+        String internalHostName = IdentityUtil.getProperty(IdentityCoreConstants.SERVER_HOST_NAME);
+        try {
```

Comment:
here code from line 244 to 256 seems to be duplicated from code in line 226 to 238.. shall we move it to a common method?

Reply:
Fixed 0c97e69